ddtrace/internal/native/__init__.py

```diff
@@ -21,39 +20,23 @@ def get_configuration_from_disk(
     """
     configurator = PyConfigurator(debug_logs)
 
-    # Set the file override if provided. Only used for testing purposes.
+    # Check if the file override is provided via environment variables
+    # This is only used for testing purposes
+    local_file_override = os.getenv("_DD_SC_LOCAL_FILE_OVERRIDE", "")
+    managed_file_override = os.getenv("_DD_SC_MANAGED_FILE_OVERRIDE", "")
     if local_file_override:
         configurator.set_local_file_override(local_file_override)
     if managed_file_override:
         configurator.set_managed_file_override(managed_file_override)
 
-    config = []
+    config = {}
     try:
-        config = configurator.get_configuration()
+        for entry in configurator.get_configuration():
+            env = entry["name"]
+            if env not in entry or "source" == "fleet_stable_config":
```

Comment:
So `local_stable_config` entries overridden by env vars will not be reported in telemetry? I think it'd be best for telemetry to use the two-dicts approach

Reply:
We currently only send telemetry for configurations that are set (not configurations that are ignored due to lower precedence). This is something can address in a follow up. I did refactor this method to use the two dict approach. This will allow us to introduce improvements.